prow/git/v2/interactor.go

```diff
@@ -164,20 +165,31 @@ func (i *interactor) Merge(commitlike string) (bool, error) {
 // MergeWithStrategy attempts to merge commitlike into the current branch given the merge strategy.
 // It returns true if the merge completes. if the merge does not complete successfully, we try to
 // abort it and return an error if the abort fails.
-func (i *interactor) MergeWithStrategy(commitlike, mergeStrategy string) (bool, error) {
+func (i *interactor) MergeWithStrategy(commitlike, mergeStrategy string, opts ...git.MergeOpt) (bool, error) {
 	i.logger.Infof("Merging %q using the %q strategy", commitlike, mergeStrategy)
 	switch mergeStrategy {
 	case "merge":
-		return i.mergeMerge(commitlike)
+		return i.mergeMerge(commitlike, opts...)
 	case "squash":
-		return i.squashMerge(commitlike)
+		return i.squashMerge(commitlike, opts...)
 	default:
 		return false, fmt.Errorf("merge strategy %q is not supported", mergeStrategy)
 	}
 }
 
-func (i *interactor) mergeMerge(commitlike string) (bool, error) {
-	out, err := i.executor.Run("merge", "--no-ff", "--no-stat", "-m", "merge", commitlike)
+func (i *interactor) mergeMerge(commitlike string, opts ...git.MergeOpt) (bool, error) {
+	args := []string{"merge", "--no-ff", "--no-stat"}
+
+	if len(opts) == 0 {
```

Comment:
This only works if the user passes a message in their opts

Reply:
Before we were hardcoding the message to `-m merge`. I though its best to keep the same thinking. Otherwise we will rely on the user to pass a message, otherwise the command will fail.

Reply:
If the user does not pass in a message via options in the way you have written this, this fails.